### PR TITLE
Add Solaris 11 support

### PR DESCRIPTION
**Pull request by nalyanyam**
Add Solaris 11 support.
The default Solaris 11 package provider (pkg) expects packages to be configured from the repository. This change also allows the legacy installation method ('sun' package provider)  to be used by specifying the package source and the admin file.

### CODE REVIEW

**Comment by ghoneycutt** [comment]
This just needs testing

**Comment by nalyanyam** [comment]
@Phil-Friderici 
Could you please ask someone from within your team to test this change that has been open for quite some time?

**Comment by Phil-Friderici** [comment]
@ghoneycutt I have asked someone else from our team in another location to give it a test drive.

File: manifests/init.pp [hunk 1]
```diff
@@ -270,6 +271,15 @@
...
   validate_re($service_ensure, '^(running|stopped)$',
     "nrpe::service_ensure must be \'running\' or \'stopped\'. Detected value is <${service_ensure}>.")
+  if $nrpe_package_provider != undef {
+    validate_string($nrpe_package_provider)
+  }
+
+  if $nrpe_package_provider != undef {
+    Package {
+      provider => $nrpe_package_provider,
+    }
```
Comment (Phil-Friderici) [inline manifests/init.pp]: you could move Package {...} into the if case above right below the validation.

**Comment by nbentoumi** [comment]
@ghoneycutt @Phil-Friderici @nalyanyam 

I did a test on Solaris 11 and working fine.

**Comment by Phil-Friderici** [comment]
@nbentoumi thanks a lot for helping us :)

File: manifests/init.pp (hunk 1 above)
Comment (nalyanyam) [inline manifests/init.pp]: Had not checked this suggestion. Will move the Package just below the validation line.
Thanks,

**Comment by nalyanyam** [comment]
Old spec tests fail for STRICT_VARIABLES="yes", with higher **gem version**. I have not understood this issue properly but suggested solutions;  disable STRICT_VARIABLES, pin to an old gem version, or fix the strict variables. Not sure which way is the best?

**Comment by Phil-Friderici** [comment]
@nalyanyam I will help you fixing these failures so that we keep the STRICT_VARIABLES test active.

**Comment by nalyanyam** [comment]
@Phil-Friderici 
Thanks for checking these spec tests.

**Comment by Phil-Friderici** [comment]
I have added PR https://github.com/ghoneycutt/puppet-module-nrpe/pull/43 to fix the issues that were introduced with Puppet 4.4.
After that got merged I guess your commit will pass the spec tests successfully.

**Comment by ghoneycutt** [comment]
Branch has conflicts

File: Gemfile [hunk 2]
```diff
@@ -27,4 +27,6 @@ gem 'puppet-lint-variable_contains_upcase'
 # rspec must be v2 for ruby 1.8.7
 if RUBY_VERSION >= '1.8.7' and RUBY_VERSION < '1.9'
   gem 'rspec', '~> 2.0'
+  # rake must be v10 for ruby 1.8.7
```
Comment (ghoneycutt) [inline Gemfile]: You can revert this as it is now upstream and a potential source of the conflict

**Comment by ghoneycutt** [comment]
This also needs spec tests

**Comment by nalyanyam** [comment]
@ghoneycutt  I have added spec tests.

File: metadata.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ghoneycutt-nrpe",
-  "version": "1.1.1",
+  "version": "1.1.2",
```
Comment (ghoneycutt) [inline metadata.json]: The maintainer of the software does this. Please revert this change.

File: spec/classes/init_spec.rb [hunk 4]
```diff
@@ -1459,17 +1272,49 @@
...
+        let :params do
+          { :nrpe_package           => 'nrpe',
+            :provider               => true,
+          }
+        end
+
+        it 'should fail' do
+          expect {
+            should contain_class('nrpe')
+          }.to raise_error(Puppet::Error)
```
Comment (ghoneycutt) [inline spec/classes/init_spec.rb]: The error string that is returned must be tested to ensure that it errored in an expected way.

File: metadata.json (hunk 3 above)
Comment (nalyanyam) [inline metadata.json]: This comes from your commit after merging @Phil-Friderici 's changes.
It appears in my initial PR probably because i updated from the upstream. Should i undo? But then i will miss the other upstream changes.

Comment (ghoneycutt) [inline metadata.json]: Ok. Nevermind.

**Comment by ghoneycutt** [comment]
@nalyanyam Once the spec test has been updated, I need sign off from someone else such as Phil or Anders that this works and looks good to merge.

**Comment by nalyanyam** [comment]
@Phil-Friderici @nbentoumi @anders-k-lindgren 
Could you please test this change and report the results here?

**Comment by nbentoumi** [comment]
@nalyanyam @ghoneycutt @Phil-Friderici 

did another test on Solaris 11 and working fine

**Comment by ghoneycutt** [comment]
Thank you everyone!

Released in v1.2.0
